.github/workflows/test262.yml

```diff
@@ -45,7 +45,7 @@ jobs:
         run: |
           cd boa
           mkdir ../results
-          cargo run --release --bin boa_tester -- run -v -o ../results/test262
+          cargo run --release --bin boa_tester -- run -O -v -o ../results/test262
```

Comment:
The enabled optimization in test262 is just for testing for now.

> At least on any optimizer updates, should we run the tests on both optimized and non-optimized in our CI to insure that an update to one doesn't break something in another?

Yes, completely agree! maybe we can configure to start test262 with optimizations when change to the `optimizer` module/directory happens (maybe reviewpad mentioned by @Razican  can be used for this, since it has a nice way of [checking if a directory contents are changed](https://docs.reviewpad.com/guides/built-ins/#hasfilepattern)). And we should probably run it on `main` push and assert that the non-optimized and optimized are equal, just to be sure :)